### PR TITLE
docs(next-usage): usage of serveractions with next

### PR DESCRIPTION
**Pull request by enyelsequeira**
# Changelog

- Usage of `NextJs` with App router and using server actions
- @crutchcorn  as discussed on discord here is the usage of `form` with server actions, let me know if updates are needed

Thank you!

### CODE REVIEW

File: docs/framework/react/guides/next-js.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+   return (
+    .....
+   )
+}
+
+```
+- Understanding `useFormState` Hook: This hook is pivotal in managing the state and actions of our form. Here, we provide it with an action `someAction` and an initial state derived from `formFactory.initialFormState`. This setup enables the component to  handle form submissions and state changes.
+
+- lets understand what is happening here, we need to use the `useFormState` hooks which give us the current state and the action. The way the hooks work is we have to give it an `action` in our case would be `someAction` and a initialState which for use would be `formFactory.initialFormState`
```
Comment (crutchcorn) [inline docs/framework/react/guides/next-js.md:115]: Capitalize `lets` :)

File: docs/framework/react/guides/next-js.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+- Benefits of `formFactory`: Much like the useForm hook, formFactory streamlines the process of form management. It provides us with necessary functionalities such as:
+  - `useStore`: Observes and reflects the current state of the form on the client side.
+  - `Provider`: Acts as a context provider for the form, ensuring state and actions are accessible throughout the component.
+  - `Subscribe`: Enables the component to listen to form-specific events, like `canSubmit` and `isSubmitting`.
+  - `handleSubmit`: Orchestrates the submission logic of the form.
+  - `Field`: Manages individual form fields, adopting the `renderProps` pattern for greater flexibility.
+
+- the `formFactory.useForm` takes a few properties that we can pass to it as an object the one we are using here is transform and we are using another hook that `@tanstack/react-form` provides which is the `useTransform` and you can see we are passing a callback and we call `mergeForm` which is a function that comes from `@tanstack/react-form` and to this we need to pass the dependecies array in our case would be the `[state]`
```
Comment (crutchcorn) [inline docs/framework/react/guides/next-js.md:124]: Let's capitalize "the" here

File: docs/framework/react/guides/next-js.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
+---
+id: next-js
+title: Next JS Usage
+---
+
+# Using TanStack Form in a Next.js App Router
+
+TanStack Form is compatible with React out of the box, supporting `SSR` and being framework-agnostic. However, specific configurations are necessary according to your chosen framework.
```
Comment (crutchcorn) [inline docs/framework/react/guides/next-js.md:8]: Let's add a link before the start of the content linking out to this blog series I wrote for the topic:

https://unicorn-utterances.com/collections/react-beyond-the-render

> Before reading this guide, it's suggested you understand how React Server Components and React Server Actions work. [Check out this blog series for more information.](https://unicorn-utterances.com/collections/react-beyond-the-render)

**Review by crutchcorn — CHANGES_REQUESTED** [review]
Minor changes, then we can merge :)

File: docs/config.json [hunk 4]
```diff
@@ -76,6 +76,10 @@
             {
               "label": "UI Libraries",
               "to": "framework/react/guides/ui-libraries"
+            },
+            {
+              "label": "Next JS",
```
[the remark this answers is not in the document]
Comment (enyelsequeira) [reply docs/config.json:81]: Done

**Comment by enyelsequeira** [comment]
hey @crutchcorn  changes you requested have been made and needed to change some files names to do the routing corrected

Thank you

**Review by crutchcorn — APPROVED** [review]

**Pull request merged**